src/core/ext/transport/binder/transport/binder_transport.cc

```diff
@@ -157,21 +157,25 @@ static void perform_stream_op(grpc_transport* gt, grpc_stream* gs,
   }
   if (op->send_message && gbs->cancellation_error.ok()) {
     gpr_log(GPR_INFO, "send_message");
-    grpc_slice s;
-    bool next_result =
-        op->payload->send_message.send_message->Next(SIZE_MAX, nullptr);
-    gpr_log(GPR_INFO, "next_result = %d", static_cast<int>(next_result));
-    op->payload->send_message.send_message->Pull(&s);
-    auto* p = GRPC_SLICE_START_PTR(s);
-    int len = GRPC_SLICE_LENGTH(s);
-    std::string message_data(reinterpret_cast<char*>(p), len);
+    size_t remaining = op->payload->send_message.send_message->length();
+    std::string message_data;
+    do {
+      grpc_slice message_slice;
+      GPR_ASSERT(
+          op->payload->send_message.send_message->Next(SIZE_MAX, nullptr));
+      op->payload->send_message.send_message->Pull(&message_slice);
+      auto* p = GRPC_SLICE_START_PTR(message_slice);
```

Comment:
Any reason to use auto here?

Reply:
It was `auto*` before this change. I can change it back to `uint8_t*`.

Reply:
Done.